packages/notify-client/src/controllers/engine.ts

```diff
@@ -336,6 +337,71 @@ export class NotifyEngine extends INotifyEngine {
     return true;
   };
 
+  public getNotificationHistory: INotifyEngine["getNotificationHistory"] =
+    async ({ topic, limit, startingAfter }) => {
+      this.isInitialized();
+
+      const subscription = this.client.subscriptions.get(topic);
+
+      const identityKey = encodeEd25519Key(
+        await this.client.identityKeys.getIdentity({
+          account: subscription.account,
+        })
+      );
+
+      const issuedAt = Math.round(Date.now() / 1000);
+      const expiry =
+        issuedAt + ENGINE_RPC_OPTS["wc_notifyGetNotifications"].res.ttl;
```

Comment:
Doesn't this technically have to be `req.ttl`? Incidentally the same value but could cause subtle bugs if they ever deviate
```suggestion
        issuedAt + ENGINE_RPC_OPTS["wc_notifyGetNotifications"].req.ttl;
```